src/main.rs

```diff
@@ -31,29 +31,74 @@ fn try_reload_config(config: &mut Config, config_file: &String) {
     }
 }
 
+///// Start parsing client input
+
 fn _parse_client_cmd(raw: &String) -> Result<String> {
     if let Some(cmd) = raw.split_whitespace().next() {
+        Ok(cmd.to_string())
+    } else {
+        Err(Error::WrongClientInputFormat)
+	}
+}
+
+fn _parse_client_arg(raw: &String) -> Result<String> {
+    if let Some(cmd) = raw.split_whitespace().skip(1).next() {
         let index = cmd.find(":");
         if index.is_some() {
-            // TODO!          // let mut split_cmd =
-  
-            Ok(cmd.to_string())
+            let split_cmd = &cmd[0..index.unwrap()];
+            println!("{:?}", split_cmd.to_string());
+            Ok(split_cmd.to_string())
         } else {
+            println!("{:?}", cmd.to_string());
             Ok(cmd.to_string())
         }
     } else {
-        Err(Error::FieldCommandIsNotSet)
+        Err(Error::WrongClientInputFormat) // repondre au client
     }
 }
 
-fn _parse_client_arg(raw: &String) -> Result<String> {
+fn _parse_client_process(raw: &String, client_arg: &String) -> Result<Option<u32>> {
     if let Some(cmd) = raw.split_whitespace().skip(1).next() {
-        Ok(cmd.to_string())
+        if let Some(index) = cmd.find(":") {
+            let split_cmd = &cmd[index + 1..];
+            if let Some(last_split_index) = split_cmd.rfind("_") {
+                let (arg, num_proc_str) = split_cmd.split_at(last_split_index);
+                let num_proc_str = &num_proc_str[1..];
+                if arg == *client_arg {
+                    if let Ok(number) = num_proc_str.parse::<u32>() {
+                        return Ok(Some(number));
+                    }
+                }
+            }
+        }
+    }
+    Ok(None)
+}
+
+fn is_job(config: &mut Config, cmd: &str) -> bool {
+    let keys = config.get_all_keys();
+    for key in keys {
+        println!("{}", key);
+        if key == cmd {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
To find if a job is in the key list you should use [this](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.contains_key)

Need to create this function in config.rs:
```rust
pub fn contains_key(&mut self, key: &String) -> bool {
    self.map.contains_key(key)
}
```

Reply:
okay !